packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -398,13 +421,35 @@ class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField>
     }
   }
 
+  void _onScroll() {
+    if (_ancestorScrollPosition == null || !_ancestorScrollPosition!.hasPixels) {
+      return;
+    }
+    final RenderBox? renderBox = context.findRenderObject() as RenderBox?;
+    final double currentHeight = renderBox?.size.height ?? 0.0;
+    setState(() { _fadeExtent = _calculateScrollOpacity(currentHeight, _maxHeight); });
+  }
+
+  static double _calculateScrollOpacity(double currentHeight, double maxHeight) {
```

Comment:
Does this assume that the search field is at the top of the scrollable content? What happens if it isn't?

Reply:
this is only useful when the search field is resizing (i.e it's height is reducing from its initial maximum height). The only reason it's called `_calculateScrollOpacity` is because this resizing in this case must happen due to a scroll (because of the if-block on line 425). So it's scroll position isn't really relevant and is used only to check that this resizing effect is happening in the midst of a scroll.

Reply:
TLDR we don't care about where the search field is in the scroll view, only that it is in a scroll view and resizing.